### PR TITLE
Add documentation for contributing to tinylog

### PR DESCRIPTION
**Pull request by pmwmedia**
### Description

Linked issue: #196 

Contributing guidelines

### CODE REVIEW

**Comment by pmwmedia** [comment]
@proohit Have I missed anything that could be helpful for new committers?

**Comment by proohit** [comment]
I will have a look at it tomorrow :)

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Before creating a pull request, please always ensure that this Maven command can be executed successfully. It compiles the code, runs JUnit tests, checks the code style via Checkstyle, and analyzes the code with SpotBugs.
+
+## IDE support
+
+In the [configuration folder](./configuration), there is a [code formatter](./configuration/formatter.xml) for Eclipse and [Checkstyle rules](./configuration/checkstyle-rules.xml), which can be used for Eclipse's and IntelliJ's Checkstyle integration for showing Checkstyle violations directly in the IDE. 
+
+## Source code
+
+tinylog 2 supports any Java version from Java 6 and later. Therefore, all source code must be compatible with Java 6. Therefore, the usage of diamond operators and lambdas is not supported.
```
Comment (proohit) [inline contributing.md]: Maybe make the diamond operators and lambdas as examples. There are many more features like some String methods, type inference etc.

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+```
+mvn clean verify
+```
+
+Before creating a pull request, please always ensure that this Maven command can be executed successfully. It compiles the code, runs JUnit tests, checks the code style via Checkstyle, and analyzes the code with SpotBugs.
+
+## IDE support
+
+In the [configuration folder](./configuration), there is a [code formatter](./configuration/formatter.xml) for Eclipse and [Checkstyle rules](./configuration/checkstyle-rules.xml), which can be used for Eclipse's and IntelliJ's Checkstyle integration for showing Checkstyle violations directly in the IDE.
```
Comment (proohit) [inline contributing.md]: Also works for VS Code with the Java language support extension :)

Comment (pmwmedia) [reply contributing.md]: I have never used VS Code for Java. Does it support the code formatter or Checkstyle rules?

File: contributing.md (hunk 1 above)
Comment (pmwmedia) [reply contributing.md]: You are right! I should clarify that I mean both as examples.

**Comment by pmwmedia** [comment]
Thank you for your review! Would you be able to build tinylog with the given information? Have I missed anything?

File: contributing.md (hunk 2 above)
Comment (proohit) [reply contributing.md]: Vs code is very extension driven. The Java language support extension supports formatter configurations.

There is also a checkstyle extension, I haven't tried it though.

Check out this link for further information https://code.visualstudio.com/docs/java/java-linting

**Comment by proohit** [comment]
Yep I would be able to build with those hints.

For the tests: Do you expect a minimum code coverage? If yes, then you could add that too.

**Comment by pmwmedia** [comment]
> For the tests: Do you expect a minimum code coverage? If yes, then you could add that too.

Good suggestion! I have just added it.

**Pull request merged**